mono/mini/tramp-amd64.c

```diff
@@ -969,6 +969,88 @@ mono_arch_create_sdb_trampoline (gboolean single_step, MonoTrampInfo **info, gbo
 
 	return buf;
 }
+
+
+#ifdef ENABLE_INTERPRETER
+/*
+ * mono_arch_get_enter_icall_trampoline:
+ *
+ *   A trampoline that handles the transition from interpreter into native world.
+ *   It requiers to set up a descriptor (MethodArguments) that describes the
+ *   required arguments passed to the callee.
+ */
+gpointer
+mono_arch_get_enter_icall_trampoline (MonoTrampInfo **info)
+{
+	guint8 *start = NULL, *code, *exits[4], *leave_tramp;
+	MonoJumpInfo *ji = NULL;
+	GSList *unwind_ops = NULL;
+	static int arg_regs[] = {AMD64_ARG_REG1, AMD64_ARG_REG2, AMD64_ARG_REG3, AMD64_ARG_REG4};
+	int i;
+
+	start = code = (guint8 *) mono_global_codeman_reserve (256);
+
+	/* save MethodArguments* onto stack */
+	amd64_push_reg (code, AMD64_ARG_REG2);
```

Comment:
This and other stack manipulation require unwind info to be generated.

Zoltan can explain how to add it.